app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -58,12 +60,7 @@ class PromotedStoryListView @JvmOverloads constructor(
       adapter = createAdapter()
     }
 
-    if (newDataList == null || newDataList.isEmpty()) {
-      Log.w(PROMOTED_STORY_LIST_VIEW_TAG, ": failed to resolve new story list data")
```

Comment:
Sorry, now I'm confused--don't we still want to log when it's null & no-op? You injected a logger & kept the flag for this purpose but didn't use it.

Reply:
[Related thread](https://github.com/oppia/oppia-android/pull/2246#discussion_r558327352) for this exact line

I don't think that think it can be null because even the binding code in the layout won't result to null -- we require this list in the PromotedStoryListViewModel (this isn't live data). If the live data used to create the view model list is null, then we don't create the view model at all.

HomeViewModel has the live data list and if it's not empty or null, it will create a PromotedStoryListViewModel with the !null and !empty list. So unless we're creating the view without a view model, this shouldn't be ever be null -- does this sound reasonable?

Reply:
Actually, I don't think that's true (strictly speaking). AS was making that suggesting because it wasn't mark as nullable. Keep in mind that databinding is a Java system, not Kotlin, so anything that's non-primitive can be null (e.g. databinding _can_ generate code that passes null here even if Kotlin isn't expecting it, and that will trigger a NPE to be thrown & crash the app).

In terms of when null can be set, databinding will always default to null if you don't explicitly specify a default (such as [here](https://github.com/oppia/oppia-android/blob/03b555379e0ade0d53447104aa27efe349a6fc4e/app/src/main/res/layout-land/onboarding_fragment.xml#L47)). In fact, it always has the potential to bind null if the view is ever used after inflation & before the binding values are set in the databinding object. We've run into NPE issues in the past because we tried accessing databinding values before they were bound & after the view was attached (I think we talked about this earlier this week: onAttachedToWindow() is called immediately before the first batch of databinding evaluations happen, and subsequent evaluations happen again whenever the bindings themselves are changed). I think given how RecyclerView & the custom adapter works, you might be right that realistically we can't expect it to be null, but in terms of contracts databinding definitely allows it.

Reply:
Done -- this makes sense to me now, and I've added the null checks

Reply:
Thanks!